mne/viz/utils.py

```diff
@@ -2847,3 +2849,208 @@ def center_cmap(cmap, vmin, vmax, name="cmap_centered"):
         for color, name in zip(cmap(old), colors):
             cdict[name].append((new, color, color))
     return LinearSegmentedColormap(name, cdict)
+
+
+def _set_psd_plot_params(info, proj, picks, ax, area_mode):
+    """Set PSD plot params."""
+    import matplotlib.pyplot as plt
+    _data_types = ('mag', 'grad', 'eeg', 'seeg', 'ecog')
+    _check_option('area_mode', area_mode, [None, 'std', 'range'])
+    picks = _picks_to_idx(info, picks)
+
+    # XXX this could be refactored more with e.g., plot_evoked
+    # XXX when it's refactored, Report._render_raw will need to be updated
+    megs = ['mag', 'grad', False, False, False]
+    eegs = [False, False, True, False, False]
+    seegs = [False, False, False, True, False]
+    ecogs = [False, False, False, False, True]
+    titles = _handle_default('titles', None)
+    units = _handle_default('units', None)
+    scalings = _handle_default('scalings', None)
+    picks_list = list()
+    titles_list = list()
+    units_list = list()
+    scalings_list = list()
+    for meg, eeg, seeg, ecog, name in zip(megs, eegs, seegs, ecogs,
+                                          _data_types):
+        these_picks = pick_types(info, meg=meg, eeg=eeg, seeg=seeg, ecog=ecog,
+                                 ref_meg=False)
+        these_picks = np.intersect1d(these_picks, picks)
+        if len(these_picks) > 0:
+            picks_list.append(these_picks)
+            titles_list.append(titles[name])
+            units_list.append(units[name])
+            scalings_list.append(scalings[name])
+    if len(picks_list) == 0:
+        raise RuntimeError('No data channels found')
+    if ax is not None:
+        if isinstance(ax, plt.Axes):
+            ax = [ax]
+        if len(ax) != len(picks_list):
+            raise ValueError('For this dataset with picks=None %s axes '
+                             'must be supplied, got %s'
+                             % (len(picks_list), len(ax)))
+        ax_list = ax
+    del picks
+
+    fig = None
```

Comment:
this is not needed or it can be changed by:
```py
fig = plt.figure() if ax is None else ax_list[0].get_figure()
```
and remove the other 2 assignations to fig.

what I don't understand is why we use `ax` to drive the if and then we use `ax_list[0]` to get the figure.

Reply:
I'll confess I simply copied that over from viz/raw.py without examining it. I wouldn't be able to get to rewriting that for several days, though if someone else does it, I would not object...

Reply:
I guess that this was what I was pointing at, that maybe now that we were refactoring this code we could also improve it a bit for the next time we come around, because as it is it requires a couple of reads through to understand what it does. (and now we have it fresh)

I'm fine to do that in a separate PR. That works for me.